gulp-tasks/utils/package-runner.js

```diff
@@ -0,0 +1,84 @@
+const path = require('path');
+const glob = require('glob');
+
+/**
+ * This methods only purpose is to call a function with
+ * the package path that needs to be acted upon.
+ *
+ * If we ran gulp as `gulp build` we would want the
+ * 'build' task to run against all packages.
+ *
+ * If we ran gulp as `gulp build --project workbox-core`
+ * we would want the 'build' task to run against `workbox-core`
+ * **only**.
+ *
+ * This method simplifies how we would write the 'build'
+ * task and should be the only way we write functions.
+ *
+ * ```javascript
+ * gulp.task('build',
+ *   gulp.series(
+ *     packageRunner(someFunction, someArg, someMoreArgs)
```

Comment:
I understand what you're trying to accomplish with `packageRunner()` in the abstract, but perhaps include an example of what `someFunction` could be? Is it a transform (e.g minification) that returns a stream?